### PR TITLE
Merge 2.8

### PR DESCRIPTION
**Pull request by wallyworld**
Merge 2.8 with these PRs:

#12753 Disregard refcount for blobstore cleanup
#12758 No panic missing bundle app body
#12759  Fix errant relative timeout for migration minion reports
#12760  Include hostname in multi-watcher machine deltas
#12750 Refactor series support to that juju itself can set its own supported series

Lots of import churn around os and series package imports, hence lots of conficts.

```
# Conflicts:
#       api/imagemetadatamanager/client_test.go
#       apiserver/facades/agent/machine/machiner.go
#       apiserver/facades/agent/provisioner/provisioninginfo.go
#       apiserver/facades/client/bundle/bundle.go
#       apiserver/facades/client/client/client.go
#       apiserver/facades/client/machinemanager/machinemanager.go
#       apiserver/facades/client/machinemanager/machinemanager_test.go
#       apiserver/facades/controller/metricsmanager/metricsmanager.go
#       cloudconfig/cloudinit/interface.go
#       cloudconfig/machinecloudconfig.go
#       cloudconfig/providerinit/providerinit.go
#       cloudconfig/providerinit/renderers/interface.go
#       cloudconfig/userdatacfg.go
#       cloudconfig/userdatacfg_unix.go
#       cloudconfig/userdatacfg_win.go
#       cmd/juju/application/deploy_test.go
#       cmd/juju/application/deployer/series_selector.go
#       cmd/juju/application/deployer/series_selector_test.go
#       cmd/juju/application/setseries.go
#       cmd/juju/commands/bootstrap_test.go
#       cmd/juju/commands/main_test.go
#       cmd/juju/commands/upgrademodel.go
#       cmd/juju/machine/upgradeseries.go
#       cmd/juju/status/formatter.go
#       cmd/jujud/agent/bootstrap.go
#       cmd/jujud/agent/bootstrap_test.go
#       cmd/jujud/run/run.go
#       cmd/jujud/run/run_test.go
#       cmd/plugins/juju-metadata/addimage.go
#       cmd/plugins/juju-metadata/imagemetadata.go
#       cmd/plugins/juju-metadata/imagemetadata_test.go
#       container/lxd/image.go
#       container/lxd/server.go
#       core/model/model.go
#       core/paths/logfile.go
#       core/paths/paths.go
#       core/series/supported.go
#       core/series/supported_test.go
#       core/series/supportedseries.go
#       core/series/supportedseries_test.go
#       environs/bootstrap/bootstrap.go
#       environs/bootstrap/bootstrap_test.go
#       environs/bootstrap/tools.go
#       environs/bootstrap/tools_test.go
#       environs/config/config_test.go
#       environs/imagedownloads/simplestreams.go
#       environs/imagedownloads/simplestreams_test.go
#       environs/imagemetadata/generate.go
#       environs/imagemetadata/simplestreams.go
#       environs/jujutest/livetests.go
#       environs/manual/winrmprovisioner/winrmprovisioner.go
#       environs/simplestreams/testing/testing.go
#       environs/sync/sync.go
#       environs/sync/sync_test.go
#       environs/testing/tools.go
#       environs/tools/marshal.go
#       environs/tools/simplestreams.go
#       environs/tools/testing/testing.go
#       environs/tools/tools.go
#       featuretests/cmd_juju_upgrade_test.go
#       go.mod
#       go.sum
#       juju/testing/conn.go
#       provider/azure/environ.go
#       provider/azure/internal/imageutils/images.go
#       provider/azure/userdata.go
#       provider/azure/vmextension.go
#       provider/cloudsigma/userdata.go
#       provider/cloudsigma/userdata_test.go
#       provider/common/bootstrap.go
#       provider/common/disk.go
#       provider/ec2/image_test.go
#       provider/ec2/userdata.go
#       provider/ec2/userdata_test.go
#       provider/gce/environ_broker.go
#       provider/gce/environ_broker_test.go
#       provider/gce/google/disk.go
#       provider/gce/userdata.go
#       provider/gce/userdata_test.go
#       provider/joyent/userdata.go
#       provider/joyent/userdata_test.go
#       provider/lxd/userdata.go
#       provider/maas/environ.go
#       provider/maas/userdata.go
#       provider/maas/userdata_test.go
#       provider/oci/environ.go
#       provider/oci/images.go
#       provider/oci/userdata.go
#       provider/openstack/local_test.go
#       provider/openstack/userdata.go
#       provider/openstack/userdata_test.go
#       provider/oracle/environ.go
#       provider/oracle/images.go
#       provider/oracle/userdata.go
#       provider/oracle/userdata_test.go
#       provider/rackspace/environ.go
#       provider/rackspace/provider_configurator.go
#       provider/vsphere/userdata.go
#       provider/vsphere/userdata_test.go
#       service/discovery.go
#       service/discovery_test.go
#       state/application.go
#       state/backups/package_test.go
#       state/cloudimagemetadata/image.go
#       state/state.go
#       state/stateenvirons/package_test.go
#       testing/base.go
#       upgrades/preupgradesteps.go
#       version/current_test.go
#       worker/authenticationworker/worker.go
#       worker/metrics/collect/manifold.go
#       worker/proxyupdater/proxyupdater.go
#       worker/uniter/paths.go
#       worker/uniter/paths_test.go
#       worker/uniter/runner/args.go
#       worker/uniter/runner/args_test.go
#       worker/uniter/runner/context/env.go
#       worker/uniter/runner/context/env_test.go
#       worker/uniter/runner/runner.go

```

## QA steps

See PRs

### CODE REVIEW

**Comment by wallyworld** [comment]
!!build!!

File: environs/simplestreams/testing/testing.go [hunk 1]
```diff
@@ -145,7 +145,10 @@ var imageData = map[string]string{
 		     "com.ubuntu.juju:12.04:amd64",
 		     "com.ubuntu.juju:12.04:arm",
 		     "com.ubuntu.juju:13.04:amd64",
-		     "com.ubuntu.juju:13.04:arm"
+		     "com.ubuntu.juju:13.04:arm",
+ 			 "com.ubuntu.juju:16.04:amd64",
+ 			 "com.ubuntu.juju:18.04:amd64",
+ 			 "com.ubuntu.juju:18.04:arm"
```
Comment (SimonRichardson) [inline environs/simplestreams/testing/testing.go:151]: This has the wrong indention.

**Review by SimonRichardson — APPROVED** [review]
Just one issue around indention, otherwise the os/series stuff looks good. @achilleasa I'm assuming that the hostname stuff is also good?

**Comment by SimonRichardson** [comment]
@wallyworld also I would have expected go.mod changes, was there any?

**Comment by wallyworld** [comment]
> @wallyworld also I would have expected go.mod changes, was there any?

Yeah, one. A new github.com/juju/os/v2 tag

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**